django/core/cache/backends/db.py

```diff
@@ -49,52 +49,59 @@ class DatabaseCache(BaseDatabaseCache):
     pickle_protocol = pickle.HIGHEST_PROTOCOL
 
     def get(self, key, default=None, version=None):
-        key = self.make_key(key, version=version)
-        self.validate_key(key)
+        return self.get_many([key], version).get(key, default)
+
+    def get_many(self, keys, version=None):
+        if not keys:
+            return {}
+
+        key_map = {}
+        for key in keys:
+            self.validate_key(key)
+            key_map[self.make_key(key, version)] = key
+
         db = router.db_for_read(self.cache_model_class)
         connection = connections[db]
         quote_name = connection.ops.quote_name
         table = quote_name(self._table)
 
         with connection.cursor() as cursor:
+            format_string = ', '.join(['%s'] * len(key_map))
             cursor.execute(
-                'SELECT %s, %s, %s FROM %s WHERE %s = %%s' % (
+                'SELECT %s, %s, %s FROM %s WHERE %s IN (%s)' % (
                     quote_name('cache_key'),
                     quote_name('value'),
                     quote_name('expires'),
                     table,
                     quote_name('cache_key'),
+                    format_string,
                 ),
-                [key]
+                list(key_map),
```

Comment:
I think this is doing what it should, but a minor suggestion, perhaps use `key_map.keys()` instead of `list(key_map)`, they are equivalent, but do not test the developer on whether a calling `list()` on a dict returns a list of keys or values. Not a blocker for my approval, but a personal preference. Take it or leave it.

Reply:
This is fine as-is. It is well known that iterating over a dictionary iterates over the keys. Also this is explicitly returning a `list` and not an iterator (as is the case with Python 3).

Try out `type({}.keys())` to see the difference.

Reply:
huh, TIL. Transitioning to Python3 from 2.7 for my personal projects and I didn't know they had changed that. Thanks for pointing it out.